FirebaseDatabase/Sources/Realtime/FWebSocketConnection.m

```diff
@@ -179,21 +230,61 @@ - (void)send:(NSDictionary *)dictionary {
     // First send the header so the server knows how many segments are
     // forthcoming
     if (dataSegs.count > 1) {
-        [self.webSocket
-            send:[NSString
-                     stringWithFormat:@"%u", (unsigned int)dataSegs.count]];
+        NSString *formattedData =
+            [NSString stringWithFormat:@"%u", (unsigned int)dataSegs.count];
+#if TARGET_OS_WATCH
+        [self
+            sendStringToWebSocket:formattedData
+                       completion:^(NSError *_Nullable error) {
+                         if (error) {
+                             FFWarn(
+                                 @"I-RDB083017",
+                                 @"Error sending initial web socket data: %@.",
```

Comment:
Do you think we need these different error messages (especially since we print the data). If not, we could move the logging to `sendStringToWebSocket` as well.

Reply:
I had it in there for debugging and I'm not sure what to expect when getting real world reports so thought it could be helpful for differentiation since I wasn't sure about the breadth of errors we may get. Happy to consolidate if you feel strongly about it though.

Reply:
I think we should able to tell the callsites apart just based on the data, and pulling the logging into `sendStringToWebSocket` would remove a couple of redundant lines in this (small) PR.

Reply:
Fixed in latest commit.